### ISSUE TITLE
Update Search.gov implementation to use different API endpoint

### ISSUE THREAD

**Issue by jilladams** [department-of-veterans-affairs/va.gov-cms#18736]
## Status
**[2024-08-15] [Fran]** Updated ticket to include splitting apart monitoring for Search and Search Typeahead to enable applying different tolerances. Slack convo [here](https://dsva.slack.com/archives/C06RWEW3S3W/p1723665031778479).
**[2024-08-13] [Fran]** Michelle M has approved this work, although it shouldn't bump critical or project work that's moving toward deadlines.

## User Story or Problem Statement
- API Endpoint Problem Statement: The Search.gov implementation has been running into monitoring issues, with large numbers of 50x errors from the Search typeahead and main search endpoints. Search.gov has asked us to use a different endpoint for our search queries: https://api.gsa.gov/technology/searchgov/v2/results/i14y

- Search and Search Typeahead Monitoring Problem Statement
Having the same monitoring tolerance for search typeahead and search is problematic in that the issues we're seeing with search typeahead don't constitute an actual search outage, and having it within regular search is causing unnecessary noise in the existing monitor, thus causing engineers/Jill/watch officers to perform unnecessary triage for something that isn't an outage.

### Endpoint Description:
Existing endpoint:  https://search.usa.gov/api/v2/results/i14y 
Suggested new endpoint: https://api.gsa.gov/technology/searchgov/v2/results/i14y

According to search.gov, we _only_ need to change the endpoint. Under the hood, these are the same and have the same parameters / will have the same responses. The new suggested endpoint will allow the Search.gov team to increase the monitoring of VA usage and understand the nature of our API calls and responses in order to better support VA.gov workflow. 

### Search Monitor Description
Create an additional monitor for only Search Typeahead and remove it from the existing Search monitor. The tolerance level should be very tolerant (engineering/Jill decision here).

## Engineering Info
[Existing search engineering documentation: search docs](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/on-site-search/engineering#architecture-scribbling-todo-make-this-formal).

There are 3 ruby modules involved with Search. [Search](https://github.com/department-of-veterans-affairs/vets-api/tree/master/lib/search), [search typeahead](https://github.com/department-of-veterans-affairs/vets-api/tree/master/lib/search_typeahead), and [search click tracking](https://github.com/department-of-veterans-affairs/vets-api/tree/master/lib/search_click_tracking). We are only changing the 'search' module.

#### Endpoint Tasks
- [x] Add a [new feature toggle](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Addinganewfeaturetoggle(backend)) named 'search_use_v2_gsa'
- [x] Add a gsa_url value to the [existing Search config](https://github.com/department-of-veterans-affairs/vets-api/blob/e059e69efaa7051ccd0990074c47115ddac30e09/config/settings.yml#L872). See below.
- [x] Override the Search::Configuration.base_path based on flipper value. See below.

#### Monitoring Tasks
- Create a new monitor for Search Typeahead
- Move the Search Typeahead from the existing Search monitor and to the new Search Typeahead monitor
- Tolerance is set at TBD

Updated settings:
```yaml
# Settings for search using api.gsa.gov
search:
  access_key: SEARCH_GOV_ACCESS_KEY
  affiliate: va
  mock_search: false
  gsa_url: https://api.gsa.gov/technology/searchgov/v2
  url: https://search.usa.gov/api/v2
```

Override Search base_path based on flipper.
```ruby
def base_path
  if Flipper.enabled?(:search_use_v2_gsa, current_user)
    "#{Settings.search.gsa_url}/results/i14y"
  else
    "#{Settings.search.url}/search/i14y"
  end
end
```

## Acceptance Criteria
- [x] Search.gov implementation on va.gov uses new endpoint, only when the feature toggle is turned on.
- [x] Tests have been updated with new URLs
- [x] Update [documentation for on-site search](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/on-site-search) as necessary 
- [x] Review search monitoring
  - [x]  update monitors to ensure they will still accurately report Search issues from our new endpoint
  - [x] Update [search monitoring docs](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/on-site-search/engineering/monitoring.md) as necessary

~Mightcouldprobablyshould be a new ticket:~
  - ~If possible to do at the same time: break apart search monitoring for the v0Search Controller from monitoring v0 Search typeahead~ Now tracked in #19256

**Comment by randimays** [comment]
There is an entry for the search endpoint in [this vets-api config file](https://github.com/department-of-veterans-affairs/vets-api/blob/2499c41fcf3081ee89190858f4852256f74628ed/config/settings.yml#L876).

**Comment by dsasser** [comment]
## Engineering Pre-finement notes:

Search.gov base url's are defined in [settings](https://github.com/department-of-veterans-affairs/vets-api/blob/e059e69efaa7051ccd0990074c47115ddac30e09/config/settings.yml) for search, search_typeahead, and search_click_tracking endpoints.

```
# Settings for search
search:
  access_key: SEARCH_GOV_ACCESS_KEY
  affiliate: va
  mock_search: false
  url: https://search.usa.gov/api/v2

# Settings for search-typeahead
search_typeahead:
  api_key: API_GOV_ACCESS_KEY
  name: va
  url: https://api.gsa.gov/technology/searchgov/v1

  # Settings for search-click-tracking
search_click_tracking:
  access_key: SEARCH_GOV_ACCESS_KEY
  affiliate: va
  mock: false
  url: https://api.gsa.gov/technology/searchgov/v2
```


Endpoint defnition:

```
lib/search/configuration.rb:
  25      def base_path
  26:       "#{Settings.search.url}/search/i14y"
  27      end
```



Tests/Betamocks

```

spec/support/vcr_cassettes/search/503.yml:
  4      method: get
  5:     uri: https://search.usa.gov/api/v2/search/i14y?access_key=TESTKEY&affiliate=va&limit=10&offset=0&query=benefits
  6      body:

spec/support/vcr_cassettes/search/504.yml:
  4      method: get
  5:     uri: https://search.usa.gov/api/v2/search/i14y?access_key=TESTKEY&affiliate=va&limit=10&offset=0&query=benefits
  6      body:

spec/support/vcr_cassettes/search/empty_query.yml:
  4      method: get
  5:     uri: https://search.usa.gov/api/v2/search/i14y?access_key=TESTKEY&affiliate=va&limit=10&offset=0&query=
  6      body:

spec/support/vcr_cassettes/search/exceeds_rate_limit.yml:
  4      method: get
  5:     uri: https://search.usa.gov/api/v2/search/i14y?access_key=TESTKEY&affiliate=va&limit=10&offset=0&query=benefits
  6      body:

spec/support/vcr_cassettes/search/invalid_access_key.yml:
  4      method: get
  5:     uri: https://search.usa.gov/api/v2/search/i14y?access_key=INVALIDKEY&affiliate=va&limit=10&offset=0&query=benefits
  6      body:

spec/support/vcr_cassettes/search/invalid_affiliate.yml:
  4      method: get
  5:     uri: https://search.usa.gov/api/v2/search/i14y?access_key=TESTKEY&affiliate=INVALID&limit=10&offset=0&query=benefits
  6      body:

spec/support/vcr_cassettes/search/last_page.yml:
  4      method: get
  5:     uri: https://search.usa.gov/api/v2/search/i14y?access_key=TESTKEY&affiliate=va&limit=10&offset=80&query=benefits
  6      body:

spec/support/vcr_cassettes/search/page_1.yml:
  4      method: get
  5:     uri: https://search.usa.gov/api/v2/search/i14y?access_key=TESTKEY&affiliate=va&limit=10&offset=0&query=benefits
  6      body:

spec/support/vcr_cassettes/search/page_2.yml:
  4      method: get
  5:     uri: https://search.usa.gov/api/v2/search/i14y?access_key=TESTKEY&affiliate=va&limit=10&offset=10&query=benefits
  6      body:

spec/support/vcr_cassettes/search/success_utf8.yml:
  4      method: get
  5:     uri: https://search.usa.gov/api/v2/search/i14y?access_key=TESTKEY&affiliate=va&limit=10&offset=0&query=benefits
  6      body:

spec/support/vcr_cassettes/search/success.yml:
  4      method: get
  5:     uri: https://search.usa.gov/api/v2/search/i14y?access_key=TESTKEY&affiliate=va&limit=10&offset=0&query=benefits
  6      body:

spec/support/vcr_cassettes/search_click_tracking/missing_parameter.yml:
  4        method: post
  5:       uri: https://api.gsa.gov/technology/searchgov/v2/clicks/?access_key=TESTKEY&affiliate=va&module_code=I14Y&position=0&query=&url=https://www.testurl.com&user_agent=testUserAgent
  6        body:

spec/support/vcr_cassettes/search_click_tracking/success.yml:
  4        method: post
  5:       uri: https://api.gsa.gov/technology/searchgov/v2/clicks/?access_key=TESTKEY&affiliate=va&module_code=I14Y&position=0&query=testQuery&url=https://www.testurl.com&user_agent=testUserAgent
  6        body:

```


❓ Should we also change the typeahead endpoint to use the https://api.gsa.gov/technology/searchgov/v2 url?

**Comment by jilladams** [comment]
Great Q, I'll ask Jim.

**Comment by jilladams** [comment]
From Search.gov: 
> The typeahead endpoint is actually already using the regular base URL, and actually also a number of other features of the [data.api.gov](http://data.api.gov/) system that the other endpoints don't use.  The typeahead endpoint is its own special beast that has not been prioritized for a while, and is definitely on our list for future performance improvements.  
 
> With respect to the 500s, we are also seeing that spike, and have put in some mitigations, but are still dealing with something hammering the API servers.  We are also in the middle of an infrastructure migration, which, when complete, should give us much better monitoring tools for dealing with the errors.

**Comment by jilladams** [comment]
Settings changes are in vets-api. 
For testing: we don't know of a way to make this conditional in vets-api. 

We cannot test this in a Tugboat. vets-api uses Review Instances, and we think we could test this in Review Instances.

We want to verify the 5 estimate once we can talk to @eselkin about how we might test in vets-api lower envs.

**Comment by acrollet** [comment]
> For testing: we don't know of a way to make this conditional in vets-api.

Unsolicited suggestion: could you make the [base_path](https://github.com/department-of-veterans-affairs/vets-api/blob/03e0130da7984b382fe17c4f23ebcfdece7e608f/lib/search/configuration.rb#L26) return value dependent on a feature toggle?

**Comment by dsasser** [comment]
> > For testing: we don't know of a way to make this conditional in vets-api.
> 
> Unsolicited suggestion: could you make the [base_path](https://github.com/department-of-veterans-affairs/vets-api/blob/03e0130da7984b382fe17c4f23ebcfdece7e608f/lib/search/configuration.rb#L26) return value dependent on a feature toggle?

Yeah good point Adrian. We did discuss putting this behind a feature toggle, but weren't sure if we could/should do that in config settings, but as you pointed out we could actually do that in the configuration class. I'll pursue this as I think it is vital to be able to roll back easily and quickly should things not work as expected.

**Comment by jilladams** [comment]
## A pontification on monitoring

### Pertinent to this ticket
We need to make sure that after changes in this ticket, our API endpoint we're calling is monitored properly, and that our existing Search monitors that involve this endpoint make sense. That's a blocker to calling the update done, bc accurate monitoring can't lag too long behind the prod changes, since this is a P1 service. So I added that in an AC here. If we opt to break it out into a new ticket, it needs to happen in the same sprint or immediately following shipping the update. Sorry I missed this in refinement.

### Bigger picture on Search
Our monitors are set up to monitor [Search APM](https://vagov.ddog-gov.com/apm/services/search/operations/rack.request/resources?dependencyMap=qson%3A%28data%3A%28telemetrySelection%3Aall_sources%29%2Cversion%3A%210%29&deployments=qson%3A%28data%3A%28hits%3A%28selected%3Aversion_count%29%2Cerrors%3A%28selected%3Aversion_count%29%2Clatency%3A%2195%2CtopN%3A%215%29%2Cversion%3A%210%29&env=eks-prod&errors=qson%3A%28data%3A%28issueSort%3ATOTAL_COUNT%29%2Cversion%3A%210%29&fromUser=false&groupMapByOperation=null&infrastructure=qson%3A%28data%3A%28viewType%3Apods%29%2Cversion%3A%210%29&logs=qson%3A%28data%3A%28indexes%3A%5B%5D%29%2Cversion%3A%210%29&panels=qson%3A%28data%3A%28%29%2Cversion%3A%210%29&resources=qson%3A%28data%3A%28visible%3A%21t%2Chits%3A%28selected%3Atotal%29%2Cerrors%3A%28selected%3Atotal%29%2Clatency%3A%28selected%3Ap95%29%2CtopN%3A%215%29%2Cversion%3A%211%29&summary=qson%3A%28data%3A%28visible%3A%21t%2Cerrors%3A%28selected%3Acount%29%2Chits%3A%28selected%3Acount%29%2Clatency%3A%28selected%3Alatency%2Cslot%3A%28agg%3A95%29%2Cdistribution%3A%28isLogScale%3A%21f%29%2CshowTraceOutliers%3A%21t%29%2Csublayer%3A%28slot%3A%28layers%3Aservice%29%2Cselected%3Apercentage%29%2ClagMetrics%3A%28selectedMetric%3A%21s%2CselectedGroupBy%3A%21s%29%29%2Cversion%3A%211%29&traces=qson%3A%28data%3A%28%29%2Cversion%3A%210%29&view=spans&start=1723666652581&end=1723670252581&paused=false) sort of globally. However, Search APM includes calls to 3 different endpoints: 
![Screenshot 2024-08-14 at 2 09 49 PM](https://github.com/user-attachments/assets/538a06ec-42c5-4fa5-ad21-07ffaf3e16e8)
1. [V0::SearchController#index](https://vagov.ddog-gov.com/apm/resource/search/rack.request/72b7d62abdbc2323?query=env%3Aeks-prod%20operation_name%3Arack.request%20service%3Asearch&env=eks-prod&fromUser=false&summary=qson%3A%28data%3A%28visible%3A%21t%2Cerrors%3A%28selected%3Acount%29%2Chits%3A%28selected%3Acount%29%2Clatency%3A%28selected%3Alatency%2Cslot%3A%28agg%3A95%29%2Cdistribution%3A%28isLogScale%3A%21f%29%2CshowTraceOutliers%3A%21t%29%2Csublayer%3A%28slot%3A%28layers%3Aservice%29%2Cselected%3Apercentage%29%2ClagMetrics%3A%28selectedMetric%3A%21s%2CselectedGroupBy%3A%21s%29%29%2Cversion%3A%211%29&traces=qson%3A%28data%3A%28%29%2Cversion%3A%210%29&start=1723667089031&end=1723670689031&paused=false): When calls to this endpoint fail, a site user cannot search. Failures to this endpoint matter a lot. Historically a quieter / more stable endpoint.
2. [V0::SearchClickTrackingController#create](https://vagov.ddog-gov.com/apm/resource/search/rack.request/61905a2daa2ff864?query=env%3Aeks-prod%20operation_name%3Arack.request%20service%3Asearch&env=eks-prod&fromUser=false&summary=qson%3A%28data%3A%28visible%3A%21t%2Cerrors%3A%28selected%3Acount%29%2Chits%3A%28selected%3Acount%29%2Clatency%3A%28selected%3Alatency%2Cslot%3A%28agg%3A95%29%2Cdistribution%3A%28isLogScale%3A%21f%29%2CshowTraceOutliers%3A%21t%29%2Csublayer%3A%28slot%3A%28layers%3Aservice%29%2Cselected%3Apercentage%29%2ClagMetrics%3A%28selectedMetric%3A%21s%2CselectedGroupBy%3A%21s%29%29%2Cversion%3A%211%29&traces=qson%3A%28data%3A%28%29%2Cversion%3A%210%29&start=1723667099855&end=1723670699855&paused=false) - When calls to this endpoint fail, clicks don't get tracked by Search.gov. We don't care that much. Historically very noisy with errors.
3. [V0::SearchTypeaheadController#index](https://vagov.ddog-gov.com/apm/resource/search/rack.request/9bbc882c60579609?query=env%3Aeks-prod%20operation_name%3Arack.request%20service%3Asearch&env=eks-prod&fromUser=false&summary=qson%3A%28data%3A%28visible%3A%21t%2Cerrors%3A%28selected%3Acount%29%2Chits%3A%28selected%3Acount%29%2Clatency%3A%28selected%3Alatency%2Cslot%3A%28agg%3A95%29%2Cdistribution%3A%28isLogScale%3A%21f%29%2CshowTraceOutliers%3A%21t%29%2Csublayer%3A%28slot%3A%28layers%3Aservice%29%2Cselected%3Apercentage%29%2ClagMetrics%3A%28selectedMetric%3A%21s%2CselectedGroupBy%3A%21s%29%29%2Cversion%3A%211%29&traces=qson%3A%28data%3A%28%29%2Cversion%3A%210%29&start=1723667060750&end=1723670660750&paused=false) - When calls to this endpoint fail, a user doesn't receive back typeahead responses from Search.gov, but search can still be completed. It's a UX nicety, but it's not mission critical. Historically very noisy with errors.

Based on [this conversation](https://dsva.slack.com/archives/C06RWEW3S3W/p1723665031778479) in the monitoring channel, if it's possible, we want to break up our monitors to be more clear about when SearchController proper is failing, vs. Typeahead noise. 

Wouldn't it be nice to just handle all of that here. It's nice to want things, Jill, but: I'd be curious to hear from the person who picks up this ticket, whether that's reasonable or unreasonable to try to do that now or cut a follow up ticket.

**Comment by jilladams** [comment]
(Also: if the monitor is set in internals, monitoring might not be affected at all. I dunno.)

**Comment by jilladams** [comment]
I went ahead and manually updated the Revision Date on this node, and set it to the value that is coming through in migration CSV, until we can work this ticket. Added a revision log. (My change may get overwritten by next migration, which is ok and may be a data point to help us understand what the migration is doing.)

**Comment by jilladams** [comment]
@SnowboardTechie given your Rails background, we thought this might be a good candidate for this sprint, after your other assigned tickets are closed. Let's talk through it when you're ready to start, for context.

**Comment by jilladams** [comment]
EOS: PRs likely to go up today.

**Comment by jilladams** [comment]
From planning: 
* The API implementation and testing is basically done / ready. 
* From ACs, this portion should break out into a separate ticket: 

> Mightcouldprobablyshould be a new ticket:

> If possible to do at the same time: break apart search monitoring for the v0Search Controller from monitoring v0 Search typeahead 

I'll stub that, since it's my brainchild. (FYI @FranECross )

**Comment by jilladams** [comment]
Now in https://github.com/department-of-veterans-affairs/va.gov-cms/issues/19256

**Comment by SnowboardTechie** [comment]
@FranECross or @jilladams with this adjustment being introduced behind a feature flag it is being requested that we have a more detailed timeline for [when the flag will be deprecated](https://github.com/department-of-veterans-affairs/vets-api/pull/18450#issuecomment-2362231964). My understanding is that we are introducing this behind a flag to be able to safely test this in production without needing to rollback if anything goes wrong. So I think we only need the flag for 1-4weeks depending on what feels appropriate to ya'll (longer gives us more time to verify in monitors that we are seeing the desired decrease in 503s and other errors). We likely want to add a ticket to this epic for the removal of this flag and the search.gov API endpoint from the codebase once ready to fully commit to this endpoint update.
How does that sound to you?

**Comment by jilladams** [comment]
In scrum, Fran / Jill agree that ~4 weeks sounds good.

**Comment by SnowboardTechie** [comment]
@jilladams sounds great, ticket for the deprecation captured here #19278

**Comment by SnowboardTechie** [comment]
https://github.com/department-of-veterans-affairs/vets-api/pull/18450 is all merged and I have successfully tested search on staging.va.gov with the feature flag both enabled and disabled. We are all set for this to deploy Monday for prod testing.

**Comment by SnowboardTechie** [comment]
This is now live on production, [feature flag](https://api.va.gov/flipper/features/search_use_v2_gsa) enabled and search is functioning as intended.

**Comment by SnowboardTechie** [comment]
@FranECross while initial testing on production went well. After a brief time we started receiving 429s due to a lack of [fwd proxy](https://dsva.slack.com/archives/CBU0KDSB1/p1727115863655309?thread_ts=1727114422.619969&cid=CBU0KDSB1) being used with the new URLs. [The feature flag](https://api.va.gov/flipper/features/search_use_v2_gsa) will need to remain disabled until this is [updated to use fwdproxy](https://depo-platform-documentation.scrollhelp.site/developer-docs/adding-a-new-external-service-integration#AddinganewExternalServiceIntegration-ForwardProxy) (probably over [here](https://github.com/department-of-veterans-affairs/devops/blob/0f0176bb80fd4230039b6fadb03e56b68bd8b368/ansible/deployment/config/fwdproxy-vagov-prod.yml#L263))

**Comment by acrollet** [comment]
be aware that the fwdproxy code has a new home here: https://github.com/department-of-veterans-affairs/vsp-platform-fwdproxy - however I think prod is still deployed from the devops repo, you'll need to check with platform devops to get the current status.

**Comment by jilladams** [comment]
@SnowboardTechie props for handling your first prod escalation like a pro! 

If I understood the Platform thread correctly today, there's a necessary step to update the AWS parameter store when we modify the endpoint. In addition to the fwdproxy updates, are you open to me scope-creeping this a bit more, to ask us to find an appropriate place to put documentation of that AWS parameter store update? Could be in code, or in product docs (somewhere in here? https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/on-site-search/engineering).

**Comment by jilladams** [comment]
Noting: we may need to ask Devops about docs to point engineers to update the fwdproxy when adding new API endpoints. We had thought the fwdproxy would understand host-level changes, vs. endpoint level changes.

**Comment by SnowboardTechie** [comment]
I received notice that the Flipper check in the configuration for this service was causing a lot of slowdowns with environment builds. This required splitting the new endpoint into a separate service.
At this time there are 3 remaining pieces to address for this to be completed:

1. https://github.com/department-of-veterans-affairs/vets-api/pull/18693 moves the new API into the new SearchGsa service. 
2. https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/pull/3120 updates the manifests for production to overwrite environment variables so that forward proxy is utilized and we avoid rate limiting with the search API.
3. 18693 from above is now open for review, once merged, this will allow for full testing on staging of the already updated manifests(only updated for staging/dev), which when successful will allow me to open https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/pull/3120 for review to deliver those manifests to production.
4. Once all above is complete we are safe to start opening the feature flag back up on production for UAC

**Comment by jilladams** [comment]
Ah! I missed that last comment yesterday. So this means, in effect, the Search APM in Datadog is gonna end up split as well, since the SearchGSA is a new resource? Or it'll still roll up to Search APM? (Happy to chat about it if I have it all wrong / talking would be easier than writing out to clarify)

**Comment by SnowboardTechie** [comment]
@jilladams the way I have this implemented it will be tagging both Search Service's errors with the same label `Search/Results` and just slight variations in the log messages to help identify which is being used if something goes wrong. This should result in the Search APM in Datadog playing well with both.

**Comment by SnowboardTechie** [comment]
Upon raising search traffic to 50% to the GSA endpoint via the forward proxy today we were once again [rate limited](https://dsva.slack.com/archives/C06RWEW3S3W/p1728068197711879). We have emailed search.gov requesting they address the limits.

**Comment by SnowboardTechie** [comment]
@jilladams I was wondering if you'd like to close this ticket out and combine the final testing with #19414 ?
This is fully implemented and functioning as intended until it hits rate limits so seems like a good combo to not leave this waiting for search.gov updates.

**Comment by jilladams** [comment]
I'm reluctant to close it until we get to 100%, just becuase it's very easy to lose track of the details / where we are with staged rollout. In the past we've cut tix for 50% > 75% > 100% --  that's annoying bookkeeping, tho. In this case: Search.gov acknowledged they have to adjust our rate limits. Would you be comfortable with us leaving this open as Blocked until we get the ack from them that they've adjusted limits? At that point, we can start the bumping up.  #19414 can happen in parallel.  (I'm in product sync with Michelle /  @FranECross , will try to see if this makes sense to them, since 19414 isn't prioritized yet, officially.)

**Comment by jilladams** [comment]
(Went ahead and added label, status update to top of ticket, and moved to "Complete Pending Integration" which is the parking lot for situations like this, til we get unblocked.)

**Comment by SnowboardTechie** [comment]
Absolutely no problem leaving this open/blocked. Especially hearing it has caused pain before, no worries.

**Comment by jilladams** [comment]
Did get the greenlight on going ahead with #19414 to pull that in. Thank you!

**Comment by jilladams** [comment]
Bumped the email thread with Jim today, since their infra seems to now be updated.

**Comment by SnowboardTechie** [comment]
With the latest email update it appears we will need to also include an API key going forward. I created a scum 16th minute item to discuss <img width="877" alt="Image" src="https://github.com/user-attachments/assets/6e71d7e1-bb1c-474a-aa9a-23eb67387796" />

**Comment by jilladams** [comment]
From scrum: 
API keys are stored in AWS secrets store. 
We might already have one, but we don't know / should check. 
If we want to add one: Platform prefers we get AWS access and manage it ourselves. 
The API key will be registered to a first name, last name, email address, and that email address user will need to be able to sign into API.data.gov with that email credential.

We need to ask @mmiddaugh: 
* Should we go ahead with this migration?
* If so, what email makes sense to use? (Hers?)

**Comment by jilladams** [comment]
From Product Sync: we're going to let this wait a little while, and will come back to it later in the quarter when some of the VA-side questions and contract questions are more settled.

**Comment by jilladams** [comment]
> The access_token is for the Search.gov API. The api_key is for the [api.data.gov](http://api.data.gov/) system, and it allows us to set custom rate limits for that system, based on the key. You need to use both because the two systems are not integrated.

**Comment by SnowboardTechie** [comment]
@jilladams no big deal either way, but I was thinking now that we know this will require further work that we should consider moving it out of Ready to Deploy and back to refinement?

**Comment by jilladams** [comment]
Yep, done.
